### PR TITLE
17157_update phone and email validation process

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#17157

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/resources/v2/namerequest.py
Comment (severinbeauvais) [inline legal-api/src/legal_api/resources/v2/namerequest.py:53]: I think best practise is to not provide a "found" or "not found" answer before checking authn/authz.

@argush3 What do you think about it?

File: legal-api/src/legal_api/resources/v2/namerequest.py
Comment (severinbeauvais) [inline legal-api/src/legal_api/resources/v2/namerequest.py:68]: To other reviewers: the changes here, which Kevin and I discussed, is to provide a response if affiliation succeeds before even checking email/phone.

This is needed because, in some cases, the UI needs to validate the NR and does not have access to the email/phone, but we can safely assume the NR is affiliated. (Eg, at start and end of a named IA.)

**Review by severinbeauvais — APPROVED** [review]
I'm OK with this but please wait for another review.

File: legal-api/src/legal_api/resources/v2/namerequest.py
Comment (argush3) [reply legal-api/src/legal_api/resources/v2/namerequest.py:53]: I think this is ok because we don't really have auth checks on this endpoint directly.  i.e. a check whether you can use the endpoint at all.  If that was the case it should block it before even querying for the NR.

But in this case I think what the current code does in this PR make sense.

**Review by argush3 — APPROVED** [review]

**Pull request merged**
